x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/scheduler/Cron.java

```diff
@@ -246,7 +252,20 @@ public class Cron implements ToXContentFragment {
     private transient boolean nearestWeekday = false;
     private transient int lastdayOffset = 0;
 
-    public static final int MAX_YEAR = Calendar.getInstance(UTC, Locale.ROOT).get(Calendar.YEAR) + 100;
+    // Restricted to 50 years as the tzdb only has correct DST transition information for countries using a lunar calendar
+    // for the next ~60 years
+    public static final int MAX_YEAR = Calendar.getInstance(UTC, Locale.ROOT).get(Calendar.YEAR) + 50;
```

Comment:
Random question, but let's assume someone already had a cron expression in their Watcher configuration that was using a date 75 years in the future. Would changing this cause that Watch to fail now that it's a lower restriction?

Reply:
No this is simply used as a short circuit for next runtime algorithm to stop it trying to calculate too far into the future. We recalculate next runtime on each watcher service start so eventually it would come in range and have its execution time calculated.

What could happen is if they had a watch 51 years in the future and never restarted the ES node for 50 years, then the watch would never run, but I think that cause should be uncommon enough that the protection this provides against long-running next runtime calculations / guarding against DST data being wrong is worth it.